pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -0,0 +1,609 @@
+		setDegraded(r.client, ctx, egw, reconcileErr,
+			fmt.Sprintf("Error reading felix configuration Name = %s, Namespace = %s, err = %s", egw.Name, egw.Namespace, err.Error()))
+		return reconcile.Result{}, err
+	}
+
+	egwVxlanPort := egressgateway.DefaultEGWVxlanPort
+	egwVxlanVNI := egressgateway.DefaultEGWVxlanVNI
+	if fc.Spec.EgressIPVXLANPort != nil {
+		egwVxlanPort = *fc.Spec.EgressIPVXLANPort
+	}
+	if fc.Spec.EgressIPVXLANVNI != nil {
+		egwVxlanVNI = *fc.Spec.EgressIPVXLANVNI
+	}
+
+	config := &egressgateway.Config{
+		PullSecrets:       pullSecrets,
+		Installation:      installation,
+		OsType:            rmeta.OSTypeLinux,
+		EgressGW:          egw,
+		EgressGWVxlanPort: egwVxlanPort,
+		EgressGWVxlanVNI:  egwVxlanVNI,
+		Provider:          installation.KubernetesProvider,
+	}
+
+	component := egressgateway.EgressGateway(config)
+	ch := utils.NewComponentHandler(log, r.client, r.scheme, egw)
+
+	if err = imageset.ApplyImageSet(ctx, r.client, variant, component); err != nil {
+		reqLogger.Error(err, "Error with images from ImageSet")
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
+		setDegraded(r.client, ctx, egw, reconcileErr,
+			fmt.Sprintf("Error with images from ImageSet Name = %s, Namespace = %s, err = %s", egw.Name, egw.Namespace, err.Error()))
+		return reconcile.Result{}, err
+	}
+
+	if err = ch.CreateOrUpdateOrDelete(ctx, component, r.status); err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Error creating / updating resource: Name = %s, Namespace = %s", egw.Name, egw.Namespace))
+		r.status.SetDegraded(operatorv1.ResourceUpdateError,
+			fmt.Sprintf("Error creating / updating resource: Name = %s, Namespace = %s", egw.Name, egw.Namespace), err, reqLogger)
+		setDegraded(r.client, ctx, egw, reconcileErr,
+			fmt.Sprintf("Error creating / updating resource Name = %s, Namespace = %s, err = %s", egw.Name, egw.Namespace, err.Error()))
+		return reconcile.Result{}, err
+	}
+
+	// Update the status of this CR.
+	egw.Status.State = operatorv1.TigeraStatusReady
+	setAvailable(r.client, ctx, egw, string(operatorv1.AllObjectsAvailable), "All objects available")
+
+	// After the resource has been created/updated, Tigerastatus needs to be set by taking the cumulative status of the
+	// available EGW resources. Lets say we create 2 EGW resources Red, Blue. Both are degraded. Now lets create 3rd resource
+	// yellow. Though yellow is reconciled and rendered successfully, Tigerastatus should still be degraded as Red, Blue are
+	// degraded. Now lets assume Blue gets updated and gets rendered properly. In this case, Tigerastatus should still be
+	// degraded as Red is unhealthy.
+	egws, err := getEgressGateways(ctx, r.client, reconcile.Request{NamespacedName: types.NamespacedName{Namespace: ""}})
+	if err != nil {
+		// Leave the status as it is and return
+		return reconcile.Result{}, nil
+	}
+
+	// Get the cumulative status of all Egress Gateway resources.
```

Comment:
Would it be possible to only do this once in total and do it outside this func in `Reconcile()`?